### PR TITLE
Components: Do not render FormTokenField label when not defined

### LINKED ISSUE #61395: Components: FormTokenField label renders even when no defined
The `<FormTokenField />` component renders HTML elements for the label part, even when the label is not defined.

* Test using the [story](https://wordpress.github.io/gutenberg/?path=/story/components-formtokenfield--default)
* Define an empty label
* Confirm, visually, there is a space in the rendered component

Screenshot 2024-05-06 at 09 05 30

* Confirm, inspecting the elements by using the browser dev tool, there are a few elements.

Screenshot 2024-05-06 at 09 06 25

### PR DESCRIPTION
**Pull request by retrofox**
## What?

Changes introduced this PR doesn't not rendering the label elements when the property value is not defined

## Why?

Because currently, the component renders element with empty content

## How?

It will render the label components only when the property is defined

## Testing Instructions

Using storybook

1. Run storybook locally

```
npm run storybook:dev
```

2. Select the default story for the TokenFormField component

`http://localhost:50240/?path=/story/components-formtokenfield--default&args=label:`

3. Confirm that when the label is empty or not defined, now the component does not add an additional empty space

before (trunk) | after (this PR)
--------|------
Screenshot 2024-05-02 at 19 46 16 | Screenshot 2024-05-02 at 19 43 42

4. You can confirm that currently (trunk) the component renders en empty `label` attribute

image

### CODE REVIEW

**Review by alexstine — APPROVED** [review]
@retrofox Looks good. Do not forget to add a components changelog entry and then this is ready to merge given checks pass.

**Comment by retrofox** [comment]
> @retrofox Looks good.

👍 

> Do not forget to add a components changelog entry and then this is ready to merge given checks pass.

I've added the entry, but the test still fails :-(

```
Please add a CHANGELOG entry to packages/components/CHANGELOG.md,
and make sure your CHANGELOG entry has a link to the current PR.
```

File: packages/components/CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,7 @@
 
 ### Enhancements
 
+-   `FormTokenField`: Hide label when not defined ([#61395](https://github.com/WordPress/gutenberg/pull/61336)).
```
Comment (alexstine) [inline packages/components/CHANGELOG.md]:
```suggestion
-   `FormTokenField`: Hide label when not defined ([#61336(https://github.com/WordPress/gutenberg/pull/61336)).
```

**Comment by alexstine** [comment]
@retrofox Just pushed a commit, let's see if that makes them pass.

File: packages/components/CHANGELOG.md [hunk 2]
```diff
@@ -10,6 +10,7 @@
 
 ### Enhancements
 
+-   `FormTokenField`: Hide label when not defined ([#61336(https://github.com/WordPress/gutenberg/pull/61336)).
```
Comment (alexstine) [inline packages/components/CHANGELOG.md]:
```suggestion
-   `FormTokenField`: Hide label when not defined ([#61336](https://github.com/WordPress/gutenberg/pull/61336)).
```

**Comment by retrofox** [comment]
Thanks 🙇‍♂️

**Pull request merged**